evaluation/timingDiagram.py

```diff
@@ -0,0 +1,217 @@
+import argparse
+import gc
+import time
+from math import floor
+from pathlib import Path
+from random import choice
+from multiprocessing import Pool as ThreadPool
+
+import requests
+from src.Intersection import intersection_log_to
+from src.SawtoothPBFT import sawtooth_container_log_to
+from src.SmartShardPeer import smart_shard_peer_log_to
+from src.api.api_util import get_plain_text
+from src.structures import Transaction
+from src.util import make_intersecting_committees_on_host
+
+# Defaults
+NUMBER_OF_TX = 20
+NUMBER_OF_COMMITTEES = 5
+INTERSECTION = 20
+NUMBER_OF_EXPERIMENTS = 1
+OUTPUT_FILE = "TimingDiagram.csv"
+EXPERIMENT_DURATION_SECS = 300
+MEASUREMENT_INTERVAL = 5
+
+# Const
+IP_ADDRESS = "localhost"
+URL_HOST = "http://{ip}:{port}"
+
+# Opens output file and writes results in it for each data point
+def make_graph_data(outfile: Path, number_of_tx: int, experiment_duration_secs: int, measurement_interval_secs: int,
+                    number_of_intersections: int, experiments: int, committees: int):
+    out = open(outfile, 'w')
+    print("Outputting to {}".format(outfile))
+    out.write("Seconds for confirmation, Number of confirmed transactions\n")
+    print("----------------------------------------------------------")
+    print("Starting experiments for transaction amount {}".format(number_of_tx))
+    avgs = get_avg_for(number_of_tx, experiment_duration_secs, measurement_interval_secs, number_of_intersections,
+                           experiments, committees)
+    print("Experiments for transaction amount {} ended".format(number_of_tx))
+    print("----------------------------------------------------------")
+    out.write("{s}, {w}\n".format(s=number_of_tx, w=avgs["throughput"]))
+    out.close()
+
+
+# Run each experiment and calc avgs
+def get_avg_for(number_of_transactions: int, experiment_duration_secs: int, measurement_interval_secs: int,
+                number_of_intersections: int, experiments: int, committees: int):
+    throughputPerE = {}
+    for e in range(experiments):
+        print("Setting up experiment {}".format(e))
+        peers = make_intersecting_committees_on_host(committees, number_of_intersections)
+        results = run_experiment(peers, experiment_duration_secs, measurement_interval_secs, number_of_transactions)
+        throughputPerE[e] = results["throughput"]
+        print("Cleaning up experiment {}".format(e))
+        del peers
+        gc.collect()
+    throughput = {}
+    # For each experiment
+    for e in range(experiments):
```

Comment:
Basically just runs through all of the experiment's results, adds them together, and then outputs them in ascending order of key